### PR TITLE
core/dutydb: add attestation data for zero committee index

### PR DESCRIPTION
**Pull request by dB2510**
Adds attestation data for zero committee index and the given slot.

category: misc
ticket: #473

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -65,7 +65,7 @@ The following table outlines which clients have produced which duties on a publi
 
 | Duty \ Client                        |                      Teku                       |                    Lighthouse                     | Lodestar | Nimbus | Vouch | Prysm |
 | ------------------------------------ | :---------------------------------------------: | :-----------------------------------------------: | :------: | :----: | :---: | :---: |
-| _Attestation_                        |                       ✅                        |                        ✅                         |    🚧    |   🚧   |  🚧   |  🚧   |
+| _Attestation_                        |                       ✅                        |                        ✅                         |    ✅    |   🚧   |  🚧   |  🚧   |
 | _Attestation Aggregation_            |                       🚧                        |                        🚧                         |    🚧    |   🚧   |  🚧   |  🚧   |
 | _Block Proposal_                     |                       ✅                        |                        ✅                         |    🚧    |   🚧   |  🚧   |  🚧   |
```
Comment (dB2510) [inline README.md:70]: haven't updated block proposal part since it isn't tested with lodestar.
if it works on simnet can we mark it as ✅ ?

File: core/dutydb/memory.go [hunk 2]
```diff
@@ -316,6 +316,22 @@ func (db *MemDB) storeAttestationUnsafe(pubkey core.PubKey, unsignedData core.Un
...
+	// This is done in order to support clients like lodestar who call GET attestation data endpoint with 0 committee_index,
+	// to optimize network calls, since attestation data object does not depends on committee_index.
+	aKeyZeroComm := attKey{
+		Slot:    int64(attData.Data.Slot),
+		CommIdx: 0,
+	}
+
+	if value, ok := db.attDuties[aKeyZeroComm]; ok {
+		if value.String() != attData.Data.String() {
+			return errors.New("clashing attestation data", z.Any("key", aKeyZeroComm))
```
Comment (corverroos) [inline core/dutydb/memory.go:329]: make this a warning rather with a `TODO: return error once we are sure attestation data does not vary with committee index`

**Review by corverroos — APPROVED** [review]
please add a test

File: README.md (hunk 1 above)
Comment (corverroos) [reply README.md:70]: yeah we should decide whether we are going to add all these clients to the compose tests, we should probably.

Comment (dB2510) [reply README.md:70]: sure, i'll add lodestar to compose tests alongside lighthouse, teku and vmock in another PR.

**Comment by corverroos** [comment]
actually, I would prefer to follow a more defensive approach.

Instead of duplicating all attestation datas in the set to committee index 0, rather only duplicate the lowest committee index to 0 (if it isn't zero itself).

**Review by corverroos — CHANGES_REQUESTED** [review]
- TL;DR: lodestar is not spec compliant, which probably results in us not supporting them
 - Lodestar's optimisation to use a single attestation data for all committees is a probable blocker ito supporting it.
 - The beacon spec provides NO guarantees that attestation datas for different committees MUST be identical. 
 - In the event that charon stores different attestation datas in the dutydb (for the same slot), then lodestar will basically attest with a SINGLE attestation data while other VCs will attest using the different datas, resulting in aggregation failure. 
 - It might be that that in 99% of the time, all attestation datas are actually the same for different committees, it is however only an accidental implementation phenomenon which might change at anypoint due to changes in implementation or race conditions in the code.
 - We are going to monitor whether we can detect inconsistent attestation data for different committees, which will prove that we cannot support lodestar.
 - If we cannot detect inconsistencies, then we are in a tough spot, since we can provisionally support lodestar with the caveat that attestation might fail at any point.
